src/groupeddataframe/grouping.jl

```diff
@@ -107,10 +107,30 @@ Compat.lastindex(gd::GroupedDataFrame) = length(gd.starts)
 Base.first(gd::GroupedDataFrame) = gd[1]
 Base.last(gd::GroupedDataFrame) = gd[end]
 
-Base.getindex(gd::GroupedDataFrame, idx::Int) =
+Base.getindex(gd::GroupedDataFrame, idx::Integer) =
     view(gd.parent, gd.idx[gd.starts[idx]:gd.ends[idx]])
-Base.getindex(gd::GroupedDataFrame, I::AbstractArray{Bool}) =
-    GroupedDataFrame(gd.parent, gd.cols, gd.idx, gd.starts[I], gd.ends[I])
+Base.getindex(gd::GroupedDataFrame, idxs::AbstractArray) =
+    GroupedDataFrame(gd.parent, gd.cols, gd.idx, gd.starts[idxs], gd.ends[idxs])
+Base.getindex(gd::GroupedDataFrame, idxs::Colon) =
+    GroupedDataFrame(gd.parent, gd.cols, gd.idx, gd.starts, gd.ends)
+
+function Base.:(==)(gd1::GroupedDataFrame, gd2::GroupedDataFrame)
+    res = gd1.parent == gd2.parent
+    ismissing(res) && return missing
+    res &&
+    (gd1.cols == gd2.cols) &&
+    (gd1.idx == gd2.idx) &&
+    (gd1.starts == gd2.starts) &&
+    (gd1.ends == gd2.ends)
```

Comment:
If I'm not mistaken, checking `cols` is enough.

We could also decide that what matters is the actual grouping, even if it's based on different columns, in which case checking `idx` and `starts` would be necessary and sufficient. But that's more work without a clear advantage.

Reply:
At https://github.com/JuliaData/DataFrames.jl/pull/1520 I force `cols` to contain integers. You could apply that change here.

Reply:
OK. So I will leave only `cols` check and after #1520 all will be guaranteed to be consistent. Right?